packages/lodestar/src/network/peers/utils.ts

```diff
@@ -65,23 +66,47 @@ export function findMissingSubnets(network: INetwork): number[] {
 }
 
 /**
+ * Disconnect useless peers when node is syncing:
+ * + Disconnect all peers that do not support sync protocols
+ * + Disconnect half of bad score sync peers, gradually we'll only have good score sync peers.
+ */
+export function syncPeersToDisconnect(network: INetwork): PeerId[] {
+  const allPeers = network.getPeers({connected: true}).map((peer) => peer.id);
+  const syncPeers = network
+    .getPeers({
+      connected: true,
+      supportsProtocols: getSyncProtocols(),
+    })
+    .map((peer) => peer.id);
+  const nonSyncPeers = allPeers.filter((peer) => !syncPeers.includes(peer));
```

Comment:
I don't want to deal with peer protocols here, that's internal libp2p data in `network.ts` so I'd just base on existing apis to determine non-sync peers